lib/checkother.cpp

```diff
@@ -1391,6 +1391,16 @@ void CheckOther::checkConstVariable()
         // Skip if address is taken
         if (Token::findmatch(var->nameToken(), "& %varid%", scope->bodyEnd, var->declarationId()))
             continue;
+        // Skip if another non-const variable is initialized with this variable
+        {
+            auto match = Token::findmatch(var->nameToken(), "%type% & %name% = %varid%", scope->bodyEnd, var->declarationId());
+            if (match)
```

Comment:
I suggest:

    const Variable *matchvar = match ? match->tokAt(2)->variable() : nullptr;
    if (matchvar && !matchvar->isConst())
        continue;

that would handle extended declarations better like `const static unsigned int & x = y;`